File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/cacheRequest/ICacheRequest.java
package org.ngbp.jsonrpc4jtestharness.rpc.cacheRequest;

import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcMethod;
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcType;

import org.ngbp.jsonrpc4jtestharness.rpc.cacheRequest.model.QueryCacheUsage;

@JsonRpcType
public interface ICacheRequest {
    @JsonRpcMethod("org.atsc.CacheRequest")
    ICacheRequest cacheRequest();

    @JsonRpcMethod("org.atsc.CacheRequestDASH")
    ICacheRequest CacheRequestDASH();

    @JsonRpcMethod("org.atsc.cacheUsage")
    QueryCacheUsage queryCacheUsage();
}

File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/cacheRequest/ICacheRequest.kt
package org.ngbp.jsonrpc4jtestharness.rpc.cacheRequest

import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcMethod
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcType
import org.ngbp.jsonrpc4jtestharness.rpc.cacheRequest.model.CacheRequest
import org.ngbp.jsonrpc4jtestharness.rpc.cacheRequest.model.QueryCacheUsage

@JsonRpcType
interface ICacheRequest {
    @JsonRpcMethod("org.atsc.CacheRequest")
    fun cacheRequest(): CacheRequest

    @JsonRpcMethod("org.atsc.CacheRequestDASH")
    fun CacheRequestDASH(): CacheRequest

    @JsonRpcMethod("org.atsc.cacheUsage")
    fun queryCacheUsage(): QueryCacheUsage
}